### PR TITLE
50 and 100 Minegeld notes support

### LINKED ISSUE #12: Add support for 50 and 100 Minegeld notes
Currently, ATMs only support 1, 5 and 10 Minegeld notes but the currency mod also has 50 and 100 Minegeld notes.

### PR DESCRIPTION
**Pull request by Wuzzy2**
Fixes #12.

ATM model 2 accepts 50 Minegeld notes and ATM model 3 accepts 50+100 Minegeld notes.

Also, the code checks if the currency mod actually does have these items and if not, automatically fixes the ATM formspecs. Because the 50/100 Minegeld notes were added in later versions of currency.

### CODE REVIEW

**Comment by Niklp09** [comment]
https://github.com/mt-mods/atm/pull/13 already exists. Can I adopt parts of your changes?

**Comment by Wuzzy2** [comment]
Ouch, we worked on the same thing at the same time rofl.

I think my PR is better because it avoids code repetition and also has support for legacy versions of currency if the 50 and 100 Mg notes are missing. (For example, Catlandia doesn't have 100 Mg note.)

Feel free to "steal". ;-)

**Comment by Niklp09** [comment]
@Wuzzy2 Please fix the luacheck warnings, I'll add translation support later.

File: forms.lua [hunk 1]
```diff
@@ -1,99 +1,146 @@
...
+local mg = function(count, str)
+	if count == 50 and mg50_exists then
+		return str
+	elseif count == 100 and mg100_exists then
+		return str
+	else
+		return ""
+	end
+end
+
```
Comment (Niklp09) [inline forms.lua]: MG 100 was added in 2018, unnecessary IMO (https://github.com/mt-mods/currency/commit/03ad637fb10a139b97d90ac73690db81942a3919)

**Review by Niklp09 — COMMENTED** [review]
everything else LGTM.

**Comment by Wuzzy2** [comment]
Not unneccary given that Catlandia still doesn't have the 100 Minegeld note. My guess is they don't want it.

**Comment by wsor4035** [comment]
> Not unneccary given that Catlandia still doesn't have the 100 Minegeld note. My guess is they don't want it.

or its just very out of date......

**Comment by Wuzzy2** [comment]
If you don't like this PR you can just shut it down and pick the other one. xD

**Comment by wsor4035** [comment]
> If you don't like this PR you can just shut it down and pick the other one. xD

i havent looked at either yet. only thing that pops initially is luacheck isnt happy on this one

**Comment by wsor4035** [comment]
ok, looked through, seems the other does have more content. perhaps maybe you and @Niklp09 can talk it over as to which one goes forward

**Comment by Niklp09** [comment]
@Wuzzy2 Can we talk on IRC?

**Comment by Wuzzy2** [comment]
luacheck done.

**Pull request merged**